extension/subprocess/process.go

```diff
@@ -0,0 +1,45 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subprocess
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+type processManager struct {
+	conf           *Config
+	logger         *zap.Logger
+	shutdownSignal chan struct{}
+}
+
+func newProcessManager(conf *Config, logger *zap.Logger) *processManager {
```

Comment:
This should ideally return the interface for the extension instead of an unexported type from this package.

Reply:
I suggest also adding a compile time check to see if the type matches the interface.

Reply:
Ok, let's return the `component.Extension` interface.